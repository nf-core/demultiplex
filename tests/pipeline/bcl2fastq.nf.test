nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "bcl2fastq"
    tag "pipeline"

    test("Bcl2Fastq") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bcl2fastq'
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 6 },
                { assert snapshot(
                        // FIXME
                        // path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz.md5"),

                        path("$outputDir/bcl2fastq/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/bcl2fastq/Stats").list(),
                    ).match("bcl2fastq") }
            )
        }

    }

}
