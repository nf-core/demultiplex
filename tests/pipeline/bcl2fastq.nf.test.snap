{
    "software_versions": {
        "content": [
            "{BCL2FASTQ={bcl2fastq=2.20.0.422}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T07:58:06+0000"
    },
    "bcl2fastq": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,e92fce7b86c6447b053d72c5cb4be89c",
            [
                "AdapterTrimming.txt:md5,48ed2b914b1246c0b5d8667525550946",
                "ConversionStats.xml:md5,8fe0f57f3f5d256a0762dba75ac62d05",
                "DemultiplexingStats.xml:md5,2047ff18f5b9107c084de06e9ff943ad",
                "DemuxSummaryF1L1.txt:md5,03e5fd0c1e3079c5f8c7b4d0501b37ff",
                "FastqSummaryF1L1.txt:md5,0c6f2d87ee183b84d1051cde9a5643d1",
                "Stats.json:md5,8e5f038b8aa9e465599d3575f930e604"
            ]
        ],
        "timestamp": "2023-10-17T07:58:06+0000"
    }
}